source/client-side-encryption/client-side-encryption.rst

```diff
@@ -649,33 +649,44 @@ FLE 2 ``CreateCollection()`` and ``Collection.Drop()``
+
+- The ``EncryptedFields`` option passed in ``collectionOptions``.
+- The value of ``AutoEncryptionOpts.EncryptedFieldsMap[<databaseName>.<collectionName>]``.
+
+If the collection namespace has an associated ``EncryptedFields``, then do the following operations. If any of the following operations error, the remaining operations are not attempted:
+
+- Create the collection with name ``EncryptedFields["escCollection"]`` using default options.
+  If ``EncryptedFields["escCollection"]`` is not set, use the collection name ``enxcol_.<collectionName>.esc``.
+  Creating this collection MUST NOT check if the collection namespace is in the ``AutoEncryptionOpts.EncryptedFieldsMap``.
+- Create the collection with name ``EncryptedFields["eccCollection"]`` using default options.
+  If ``EncryptedFields["eccCollection"]`` is not set, use the collection name ``enxcol_.<collectionName>.ecc``.
+  Creating this collection MUST NOT check if the collection namespace is in the ``AutoEncryptionOpts.EncryptedFieldsMap``.
+- Create the collection with name ``EncryptedFields["ecocCollection"]`` using default options.
+  If ``EncryptedFields["ecocCollection"]`` is not set, use the collection name ``enxcol_.<collectionName>.ecoc``.
+  Creating this collection MUST NOT check if the collection namespace is in the ``AutoEncryptionOpts.EncryptedFieldsMap``.
+- Create the collection ``collectionName`` with ``collectionOptions`` and the option ``encryptedFields`` set to the ``EncryptedFields``.
 - Create the the index ``{"__safeContent__": 1}`` on collection ``collectionName``.
 
+Drivers MUST have a BSON document option named ``EncryptedFields`` in ``Collection.Drop()``.
+
+A call to a driver helper ``Collection.Drop(dropOptions)`` must check if the collection namespace (``<databaseName>.<collectionName>``) has an associated ``EncryptedFields``. Check for an associated ``EncryptedFields`` from the following:
+
+- The ``EncryptedFields`` option passed in ``dropOptions``.
+- The value of ``AutoEncryptionOpts.EncryptedFieldsMap[<databaseName>.<collectionName>]``.
+- Run a ``listCollections`` command on the database ``databaseName`` with the filter ``{ "name": "<collectionName>" }``. Check the returned ``options`` for the ``encryptedFields`` option. The value of ``encryptedFields`` is the ``EncryptedFields``.
 
-A call to a driver helper ``Collection.Drop()`` must check if the collection namespace (``<databaseName>.<collectionName>``) is in ``AutoEncryptionOpts.EncryptedFieldConfigMap``.
-If it is, then do the following operations. If any of the following operations error, the remaining operations are not attempted:
+If the collection namespace has an associated ``EncryptedFields``, then do the following operations. If any of the following operations error, the remaining operations are not attempted:
```

Comment:
Only do this if AutoEncryptionOpts.EncryptedFieldsMap is not null?

Reply:
Oof, good catch. ``listCollections`` is intended to only be run if ``AutoEncryptionOpts.encryptedFieldsMap`` is not null. Updated the step for ``listCollections``.